### DISCUSSION TITLE
05/25 피어세션

### DISCUSSION THREAD

**Discussion by joochanhyeong** [bcaitech1/p4-dkt-DKTJHGSD#25] [Meeting Notes]
< TMI > 
사다리를 탄다. 
첫빠따 : 진소정
면접 컨설팅을 받고왔다. 시간당 32만원을 받아먹더라. 좀 아닌가 싶었다(원하는대로 안해줌. 강점 알려달라니까 취약점 분석해줌). 왜 돈을 쉽게 벌지? 갑자기 술이 떠오르는데 술을 좋아한다. 요즘 위스키에 꽂혔다. 옥수수 베이스 위스키가 쩐다.

2빠따 : 이준호 
미국에서 오래 살고왔다. 대화에 있어서 차질이 있다는 것에 밑밥을 미리 깔겠다. 원활한 의사소통을 위해 노력하겠다. 거주지는 서부 -> 고~대학교는 중부 텍사스 쪽에서 살았다. 서부가 쾌적한데 물가나 경제적으로는 중부가 나았다. 

3빠따 : 이태환
혼잣말을 한다. 혼잣말 잘한다. 혼잣말보고 놀라지 않도록 양해 부탁한다.

4빠따: 이수연
이사 많이다님. 노원구 탄생. 동두천 유치원. 목동 초등학교. 충남 아산 온양온천 유명한데 중학교. 군포 고등학교. 대학교는 서울. 다양한 로컬을 경험.

6빠따: 백서윤
요리를 못하는 것도 아닌데, 가장 기초적으로 느끼는 라면 끓이는걸 매번 실패한다. 동생 피드백 왈 계량기를 써라. 계량기 썼는데도 대 실 패.

< 깃 setup >

[이준호]
- Projects에서 4가지 tab으로 진행상황 구성. 
- 단순히 멘트 참조만 가능하지 연관 코드 참조는 불가능. --> 타협점을 어떻게 논의할 수 있을까?
- Automated Kanban 템플릿 기반에서 project 생성 권장. 
- 여러 프로젝트를 왔다갔다 참조하는 것보다 주 단위 같이 비교적 기간 내 참조 형식이 좋아보임.
- 피어세션 기준 주 단위로 구분해서 이슈나 아이디어들을 기반으로 projects를 관리하자.
- 버그 관련 이슈는 깃허브 issues 게시판을 적극활용해서 효율적인 트러블슈팅을 추진하자. 
- Milestone과 issue close 관련된 이야기는 추가 합의가 필요.
- In progress 탭에서는 issue로 보내고 assign을 자기로 넣어주면 내가 이 이슈에 대해서 담당해보겠다라고 의도 표출 가능.
- Discussion 탭에서 submission result와 description 공유 환경 구축. 

[이태환]
- 실제 코드 update는 마스터 브랜치. 
- 별도의 자기 자신만의 브랜치를 생성. (현 시점 마스터에서 파생)
- 브랜치를 마스터에서 자기 자신 것으로 이동(checkout)
- Add,commit 까지는 로컬 브랜치 영역. 마스터에 영향 x
- Push 이후엔 현 마스터에 갱신.

[이수연]
- Git status로 갱신사항 확인(마스터브랜치 기준 새로운 파일 혹은 기존 파일 내 갱신된 내용)
- 현 commit 포맷에 맞게 편집하려면 git commit 후, esc+ :wq
- Git push origin <local_branch> : 원격 로컬 브랜치 add/commit 내용 원래의 브랜치로 추가.
- 이 갱신내용이 마스터 브랜치 단위로 통합해야할 시, 웹사이트 gui에서 pull request 진행.
- 해당 메시지가 합의를 거쳐야 하는 경우, pull request message에서 files change 탭에서 변경 내용 확인 통해 리뷰 가능. 특정 코드 섹션에다 리뷰 코멘트 작성 가능. 
- 최종적으로 confirm merge 단계에 다다르면 마스터 브랜치까지 갱신 완료. 
- Git pull master origin 명령어로 merge로 갱신된 마스터 브랜치 코드 로컬로 다운로드 가능. 

< 차후 mission >

[이준호]
- Trainer를 학습 관련에 필요한 것만 abstract class화시키는 방향으로 리팩토링을 진행해야하지 않을까?

[진소정]
- Research는 베이스라인 코드 분석 후 진행하는 것이 나아보인다.

[백서윤&이수연]
- EDA가 초기단계에서 방향성을 잡는데 중요하니 기존 역할 수행.
- EDA issue에서 comment에 추가적인 아이디어 공유 가능.

- Wandb 공유에 대해서
 1) Github submission 게시판에 wandb experiment index를 일치시켜서 상대의 실험결과를 공유하는데 용이하게 하자. 
 2) "#num-experiment_name” 식으로 wandb process naming

< 즉석 실험 >

- 평균 유저 당 풀이 문항이 평균 300개 인 것을 고려하여, Max len # 200으로 변경 후 default model(lstm, lstm-atten, bert) --> 전부 metric 전반적으로 하락. : 어떻게 분석해야할까?

**Comment by junho621** [comment]
와우 감사합니다~ 진짜 이렇게 정리하니깐 그날 그날 무슨 대회했는지 복기(?, 회상?, 기억?, )되고 좋네요~!
